airflow/www/views.py

```diff
@@ -1893,6 +1910,70 @@ def gantt(self, session=None):
             root=root,
         )
 
+    @expose('/extra_links')
+    @login_required
+    @wwwutils.action_logging
+    def extra_links(self):
+        """
+        A restful endpoint that returns external links for a given Operator
+
+        It queries the operator that sent the request for the links it wishes
+        to provide for a given external link name.
+
+        API: GET
+        Args: dag_id: The id of the dag containing the task in question
+              task_id: The id of the task in question
+              execution_date: The date of execution of the task
+              redirect_to: The name of the link reference to find the actual URL for
+
+        Returns:
+            200: {url: <url of link>, error: None} - returned when there was no problem
+                finding the URL, and the URL is whitelisted
+            403: {url: None, error: <error message>} - returned when the URL is not
+                whitelisted
+            404: {url: None, error: <error message>} - returned when the operator does
+                not return a URL
+        """
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        redirect_to = request.args.get('redirect_to')
+        dttm = airflow.utils.timezone.parse(execution_date)
+        dag = dagbag.get_dag(dag_id)
+
+        if not dag or task_id not in dag.task_ids:
+            flash(
+                "Task [{}.{}] doesn't seem to exist"
+                " at the moment".format(dag_id, task_id),
+                "error")
+            return redirect(request.referrer or '/admin/')
```

Comment:
`flash` and `redirect` should not be part of an ajax api

Reply:
fixed. Thanks for the catch :)